content/git/concepts/revert/revert.md

```diff
@@ -0,0 +1,72 @@
+---
+Title: 'Revert'
+Description: 'The git revert command is used to create a new commit that will undo any changes made to a previous commit.'
+Subjects:
+  - 'Bash/Shell'
+  - 'Developer Tools'
+Tags:
+  - 'Git'
+  - 'Version Control'
+CatalogContent:
+  - 'learn-the-command-line'
+  - 'learn-git'
+---
+
+In Git, the **`revert`** command is used to create a new commit that will undo any changes made to a previous commit. This can be considered as the safe undo command because it keeps previous history. The command is best used when working with others and there's a need to preserve the commit history.
+
+> **Note:** This command can be mistaken for the [`reset`](https://www.codecademy.com/resources/docs/git/reset) command, which is a more dangerous undo command because it alters existing hitory.
+
+## Syntax
+
+```shell
+git revert <commit-reference>
+```
+
+The `commit-reference` is the unique hash of a commit that is generated after creation. This hash is a long string that is a mix of characters and numbers that is usually represented by a shorter version: `a63b1329066c8ddd95c8d7bb201bacfb8b18e167` -> `a63b132`
+
+`git revert` can be used with the commit hash (as seen below) or with the `HEAD` keyword (as seen below), which refers to the commit being viewed on the currently checked-out branch.
+
+## Example
+
+Lets say there are four commits with different files being added for each commit. To view the commits, use `git reflog`
+
+```shell
+$ git reflog
+0caf1ae (HEAD -> master) HEAD@{0}: commit: Add file 4
+ffd3d9c HEAD@{1}: commit: add file 3
+f85ef36 HEAD@{2}: commit: add file 2
+fc3980d HEAD@{3}: commit (initial): add file 1
+```
+
+The above shows that each commit has a new file added. file 1 is the first commit and file 4 is the most recent commit. 
+
+The writer for file 2 mentions that file 2 is no longer needed. By looking at the log, it shows all of the commit-references. Since file 2 is the one that needs to be reverted, use the hash that is seen in the first column to remove file 2 while preserving all the other commits.
+
+```shell 
+$ git revert f85ef36
+```
+
+Another option to remove file 2 is with the `HEAD` keyword. Looking at the same log as before, it shows the commit for file 2 is located at `HEAD@{2}`. File 2 can then be removed with the following line.
+
+```shell
+$ git revert HEAD~2
+```
+
+By running the `git revert` command, this will then open a prompt to edit the message before comitting the revert. Once the message has been saved, check the logs to view the status of file 2.
```

Comment:
```suggestion
By running the `git revert` command, this will then open a prompt to edit the message before committing the revert. Once the message has been saved, check the logs to view the status of file 2:
```